### PR TITLE
feat: adapt to rst

### PR DESCRIPTION
**Pull request by 12rambau**
Adapt the indentation level of what's written to the initial indent of the opening tag. It should change nothing to markdown input.

**What**:
The table and badge created by the CLI are raw HTML. They cannot be used in .rst fie because RST files are sensible to indentation. To be compatible the tool should get the indentation of the first tag and apply it to the rest of the content (as a global offset). #300 

**Why**:
To make the tool compatible with .rst file => every doc created with Sphinx 

**How**:
I'm getting the index of the last `\n` before the start Tag and I then replace every `\n`of the newcontent with an appropriate number of extra spaces. 

**Checklist**:

- [ ] Documentation
- [ ] Tests
- [ ] Ready to be merged 
- [ ] Added myself to contributors table 

**Comments**
I don't want/can add nodes dependencies to my project (that's why I was interested by the bot) so I can't test this functionality myself. I've authorized the edit by maintainer but let me know if I need to change things on my side.

### CODE REVIEW

**Review by Berkmann18 — COMMENTED** [review]
I tried pushing changes but it blocked me so I'll just write the changes as suggestions.

File: src/generate/index.js [hunk 1]
```diff
@@ -24,11 +24,13 @@ function injectListBetweenTags(newContent) {
     ) {
       return previousContent
     }
+    const startIndent = max(0, previousContent.LastIndexOf('\n', startOfOpeningTagIndex))
+    const nbSpaces = startOfOpeningTagIndex - min(startOfOpeningTagIndex, startIndent)
```
Comment (Berkmann18) [inline src/generate/index.js:28]:
```suggestion
    const startIndent = Math.max(0, previousContent.lastIndexOf('\n', startOfOpeningTagIndex))
    const nbSpaces = startOfOpeningTagIndex - Math.min(startOfOpeningTagIndex, startIndent)
```

File: src/generate/index.js [hunk 2]
```diff
@@ -24,11 +24,13 @@ function injectListBetweenTags(newContent) {
...
     return [
       previousContent.slice(0, endOfOpeningTagIndex + closingTag.length),
       '\n<!-- prettier-ignore-start -->',
       '\n<!-- markdownlint-disable -->',
-      newContent,
+      newContent.replace('\n', '\n'+ ' '.repeat(nbSpaces)),
```
Comment (Berkmann18) [inline src/generate/index.js:33]:
```suggestion
      newContent.replace('\n', '\n' + ' '.repeat(nbSpaces)),
```

File: src/generate/index.js [hunk 3]
```diff
@@ -84,10 +86,12 @@ function replaceBadge(newContent) {
     ) {
       return previousContent
     }
+    const startIndent = max(0, previousContent.LastIndexOf('\n', startOfOpeningTagIndex))
+    const nbSpaces = startOfOpeningTagIndex - min(startOfOpeningTagIndex, startIndent)
```
Comment (Berkmann18) [inline src/generate/index.js:90]:
```suggestion
    const startIndent = Math.max(0, previousContent.lastIndexOf('\n', startOfOpeningTagIndex))
    const nbSpaces = startOfOpeningTagIndex - Math.min(startOfOpeningTagIndex, startIndent)
```

File: src/generate/index.js [hunk 4]
```diff
@@ -84,10 +86,12 @@ function replaceBadge(newContent) {
...
     return [
       previousContent.slice(0, endOfOpeningTagIndex + closingTag.length),
       '\n',
-      newContent,
+      newContent.replace('\n', '\n'+ ' '.repeat(nbSpaces)),
```
Comment (Berkmann18) [inline src/generate/index.js:94]:
```suggestion
      newContent.replace('\n', '\n' + ' '.repeat(nbSpaces)),
```

**Review by Berkmann18 — CHANGES_REQUESTED** [review]
The tests need to be updated as well (one of the updates would require a local command, unless you fancy manually updating snapshots):
image

**Comment by 12rambau** [comment]
> I tried pushing changes but it blocked me so I'll just write the changes as suggestions.

this is weird as I set this: 

Capture d’écran 2021-06-29 à 11 52 37

boh it will just be slowlyer from my side, I'll review them AFAP

**Comment by 12rambau** [comment]
according to the image you send it seems that something is not running the way I would like, I'm trying to install the yarn project locally but I can't manage to make it work. 

I've cloned my brnach on my local computer and run : `yarn init` in my fake project. Then to install the lib I ran: `yarn add file:/Users/pierrickrambaud/Documents/travail/FAO/app_buffer/all-contributors-cli` which seems to work as well. 

When I check `node_modules/.bin/`, `all-contributor` is nowhere to be found, is it normal ?

**Comment by Berkmann18** [comment]
Hmm, if you wanted to install the cli you should have done either:
```
npm i all-contributors-cli
```
or
```
yard add all-contributors-cli
```
Well, assuming you wanted to install it in a project and not as a dev/prod dependency.

**Comment by 12rambau** [comment]
Sorry for the very late answer but i had no time to dedicate to this PR these past month. 
So as I explained I would like to test the modifications I made on a fake project but i don't understand how node_module works. 

how can I test the cli in "dev" mode ?

**Comment by Berkmann18** [comment]
@12rambau That's fine, I was also very busy with other stuff.
The node_modules contain packages installed by NPM with `npm i`.

**Comment by 12rambau** [comment]
i ran `npm i` which seemed to worked (I'll ignore deprecation warnings) and then `npm t` that led to error in the snapshots as you mentioned. 

Now if I want to use it (the local version from my computer) in a fake folder (to check waht happens) what should I do ? 
If I run `npm start` from the cli folder, I get the following error: 
```
sh: ./dist/cli.js: No such file or directory
```

And obviously the `all-contributor` cli is not available. 
I'm srry if you think that my questions are trivial but there is not contributing section in the doc. i'll be happy to create one onc I finally mange to run it once

**Comment by Berkmann18** [comment]
> If I run npm start from the cli folder, I get the following error:

Have you tried running it from the root of the repo?

**Comment by 12rambau** [comment]
I get the following error : 
```
> all-contributors-cli@0.0.0-semantically-released start /Users/pierrickrambaud/Documents/travail/FAO/app_buffer/all-contributors-cli
> ./dist/cli.js

sh: ./dist/cli.js: No such file or directory
npm ERR! code ELIFECYCLE
npm ERR! syscall spawn
npm ERR! file sh
npm ERR! errno ENOENT
npm ERR! all-contributors-cli@0.0.0-semantically-released start: `./dist/cli.js`
npm ERR! spawn ENOENT
npm ERR!
npm ERR! Failed at the all-contributors-cli@0.0.0-semantically-released start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/pierrickrambaud/.npm/_logs/2022-01-12T07_55_41_479Z-debug.log
```

The use case is trivial, I've pulled the repository to my local computer, created a fake repository with nothing in it but a `README.md` file. I just want to start the CLI locally (using my branch and not the distant package) and check the output of the commands in this very repository. Once done I'll be able to modify the snapshot and finish this PR. 

I'm sure it's 101 developing procedure. can you write down the steps to reproduce it from my end ?

**Comment by tenshiAMD** [comment]
@12rambau I've fixed the checks. do you mind adding some tests? Thanks.

**Comment by 12rambau** [comment]
first thank you very much for taking over the review of this PR ! 
I would be glad to help but as long as I don't understand how to start it locally I won't be able to do anything. If you look at my previous exchange you will realize that I barely understand how I'm supposed to use this tool locally (I'm actually interested by the bot in the first place). I let you decide if it's more time-consuming for you to explain to me or to write the tests yourself.

**Comment by tenshiAMD** [comment]
@Berkmann18 I tested both .md and .rst are working

**Review by tenshiAMD — APPROVED** [review]

**Comment by tenshiAMD** [comment]
@all-contributors please add @12rambau for code

**Comment by tenshiAMD** [comment]
> I'll wait until it's available via the bot to test it in my Python projects

@12rambau you can try now

**Pull request merged**